src/terrama2/impl/DataAccessorFile.cpp.orig

```diff
@@ -0,0 +1,272 @@
+    if(!isValidDataSetName(getMask(dataSet), filter, name))
+      continue;
+
+    // creates a DataSource to the data and filters the dataset,
+    // also joins if the DCP comes from separated files
+    std::shared_ptr<te::da::DataSource> datasource(te::da::DataSourceFactory::make(dataSourceType()));
+    std::map<std::string, std::string> connInfo;
+
+    connInfo["URI"] = typePrefix() + dir.absolutePath().toStdString() + "/" + name;
+    datasource->setConnectionInfo(connInfo);
+
+    //RAII for open/closing the datasource
+    OpenClose<std::shared_ptr<te::da::DataSource> > openClose(datasource);
+
+    if(!datasource->isOpened())
+    {
+      QString errMsg = QObject::tr("DataProvider could not be opened.");
+      TERRAMA2_LOG_ERROR() << errMsg;
+
+      continue;
+    }
+
+    // get a transactor to interact to the data source
+    std::shared_ptr<te::da::DataSourceTransactor> transactor(datasource->getTransactor());
+
+    // Some drivers use tha base name and other use filename with extension
+    std::string dataSetName;
+    std::vector<std::string> dataSetNames = transactor->getDataSetNames();
+    auto itBaseName = std::find(dataSetNames.cbegin(), dataSetNames.cend(), baseName);
+    auto itFileName = std::find(dataSetNames.cbegin(), dataSetNames.cend(), name);
+    if(itBaseName != dataSetNames.cend())
+      dataSetName = baseName;
+    else if(itFileName != dataSetNames.cend())
+      dataSetName = name;
+    //No valid dataset name found
+    if(dataSetName.empty())
+      continue;
+
+    if(first)
+    {
+      //read and adapt all te:da::DataSet from terrama2::core::DataSet
+      converter = getConverter(dataSet, std::shared_ptr<te::da::DataSetType>(transactor->getDataSetType(dataSetName)));
+      series.teDataSetType.reset(static_cast<te::da::DataSetType*>(converter->getResult()->clone()));
+      assert(series.teDataSetType.get());
+      completeDataset = std::make_shared<te::mem::DataSet>(series.teDataSetType.get());
+      first = false;
+    }
+
+    assert(converter);
+    std::unique_ptr<te::da::DataSet> datasetOrig(transactor->getDataSet(dataSetName));
+    assert(datasetOrig);
+    std::shared_ptr<te::da::DataSet> teDataSet(te::da::CreateAdapter(datasetOrig.release(), converter.get(), true));
+
+    //FIXME: Nor working with raster!!
+
+    completeDataset->copy(*teDataSet);
+
+    if(completeDataset->isEmpty())
+    {
+      QString errMsg = QObject::tr("No data in dataset: %1.").arg(dataSet->id);
+      TERRAMA2_LOG_WARNING() << errMsg;
+    }
+  }// for each file
+
+  if(!completeDataset.get())
+  {
+    QString errMsg = QObject::tr("No data in dataset: %1.").arg(dataSet->id);
+    TERRAMA2_LOG_ERROR() << errMsg;
+    throw terrama2::core::NoDataException() << ErrorDescription(errMsg);
+  }
+
+  std::shared_ptr<SyncronizedDataSet> syncDataset(new SyncronizedDataSet(completeDataset));
+  series.syncDataSet = syncDataset;
+  return series;
+}
+>>>>>>> bde9b3493b10ed6e898730aa9757eb79457fbd01
```

Comment:
you added a file generate by a git conflict,
you can add only modified files with:
git add -u